diff -ruN linux-stable_20250916.orig/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcmsdh.c linux-stable_20250916/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcmsdh.c
--- linux-stable_20250916.orig/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcmsdh.c	2025-09-15 04:44:44.000000000 -0700
+++ linux-stable_20250916/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcmsdh.c	2026-03-15 07:24:51.554557897 -0700
@@ -946,11 +946,9 @@
 		goto out;
 
 	/* try to attach to the target device */
-	sdiodev->bus = brcmf_sdio_probe(sdiodev);
-	if (!sdiodev->bus) {
-		ret = -ENODEV;
+	ret = brcmf_sdio_probe(sdiodev);
+	if (ret)
 		goto out;
-	}
 	brcmf_sdiod_host_fixup(sdiodev->func2->card->host);
 out:
 	if (ret)
diff -ruN linux-stable_20250916.orig/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c linux-stable_20250916/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
--- linux-stable_20250916.orig/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c	2025-09-15 04:44:44.000000000 -0700
+++ linux-stable_20250916/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c	2026-03-15 07:26:40.281929399 -0700
@@ -4652,7 +4652,7 @@
 	return fwreq;
 }
 
-struct brcmf_sdio *brcmf_sdio_probe(struct brcmf_sdio_dev *sdiodev)
+int brcmf_sdio_probe(struct brcmf_sdio_dev *sdiodev)
 {
 	int ret;
 	struct brcmf_sdio *bus;
@@ -4754,11 +4754,12 @@
 		goto fail;
 	}
 
-	return bus;
+	return 0;
 
 fail:
 	brcmf_sdio_remove(bus);
-	return NULL;
+	sdiodev->bus = NULL;
+	return ret;
 }
 
 /* Detach and free everything */
diff -ruN linux-stable_20250916.orig/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.h linux-stable_20250916/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.h
--- linux-stable_20250916.orig/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.h	2025-09-15 04:44:44.000000000 -0700
+++ linux-stable_20250916/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.h	2026-03-15 07:21:49.075612710 -0700
@@ -389,7 +389,7 @@
 int brcmf_sdiod_probe(struct brcmf_sdio_dev *sdiodev);
 int brcmf_sdiod_remove(struct brcmf_sdio_dev *sdiodev);
 
-struct brcmf_sdio *brcmf_sdio_probe(struct brcmf_sdio_dev *sdiodev);
+int brcmf_sdio_probe(struct brcmf_sdio_dev *sdiodev);
 void brcmf_sdio_remove(struct brcmf_sdio *bus);
 void brcmf_sdio_isr(struct brcmf_sdio *bus, bool in_isr);
 
